internal/bson/bson.go

```diff
@@ -59,7 +59,7 @@ import (
 type bsontype interface {
 	bsontype() // seal for go-sumtype
 
-	ReadFrom(*bufio.Reader) error
+	ReadFrom(*bufio.Reader, int) error
```

Comment:
```suggestion
	// good explanatory comment
	readNested(*bufio.Reader, int) error
	ReadFrom(*bufio.Reader) error
```

I think we should not leak that detail into the exported API. `wire` package should not change. `bson_test.go` probably too